torch/_inductor/compile_fx.py

```diff
@@ -2116,16 +2136,43 @@ def bw_compiler(
 
         bw_compiler = SerializableAOTDispatchCompiler(OutputCode, bw_compiler)
 
-        fake_mode = detect_fake_mode(
-            example_inputs_
-        ) or torch._subclasses.FakeTensorMode(allow_non_fake_inputs=True)
+        # [See NOTE] Fake mode in inductor
+        fake_mode = detect_fake_mode(example_inputs_)
+
+        if (
+            V.aot_compilation
+            and isinstance(model_, GraphModule)
```

Comment:
Do we need this checking?

Reply:
> Do we need this checking?

you mean `isinstance(model_, GraphModule)`? It's ok to not check it, since `model_` should always be a GraphModule if we are using AOT compilation. I just added it to be extra safe. I can remove this check as well. 

We do need the rest of the checking statement though, because in JIT inductor, we should use the fake mode from the tracing context.